src/virtualenv/info.py

```diff
@@ -22,7 +23,9 @@ def get_default_data_dir():
 
     global _DATA_DIR
     if _DATA_DIR is None:
-        _DATA_DIR = Path(user_data_dir(appname="virtualenv", appauthor="pypa"))
+        key = str("_VIRTUALENV_OVERRIDE_APP_DATA")
+        folder = os.environ[key] if key is os.environ else user_data_dir(appname="virtualenv", appauthor="pypa")
```

Comment:
```suggestion
        folder = os.environ[key] if key in os.environ else user_data_dir(appname="virtualenv", appauthor="pypa")
```

additionally, usually find it better to have the empty environ variable equivalent to missing -- so maybe `os.environ.get(key)` instead?

Reply:
I think you're.  We'll have one file to bootstrap all python versions we support, from all python versions we support. In the current iteration, this is 2.7 and 3.4-3.9.